### PR TITLE
Adds plugin compatibility

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Adds a note about plugin compatibility.

### Checklist
- [x ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Thanks @kolchfa-aws this might need another update just for 2.4.0.

We have 4 plugins added just in 2.4.0.
https://github.com/opensearch-project/neural-search
https://github.com/opensearch-project/security-analytics
https://github.com/opensearch-project/security-analytics-dashboards-plugin/
https://github.com/opensearch-project/dashboards-search-relevance

**Comment by peterzhuamazon** [comment]
There is also another file for dashboards plugin.

File: _dashboards/install/plugins.md [hunk 1]
```diff
@@ -20,25 +20,31 @@ If you don't want to use the all-in-one installation options, you can install th
...
+| Reports | [dashboards-reports](https://github.com/opensearch-project/dashboards-reports) | 1.0.0 |
+| Security Analytics | [security-analytics-dashboards-plugin](https://github.com/opensearch-project/security-analytics-dashboards-plugin)| 2.4.0 |
+| Security | [security-dashboards-plugin](https://github.com/opensearch-project/security-analytics-dashboards-plugin) | 1.0.0 |
 
 ## Prerequisites
 
 - A compatible OpenSearch cluster
 - The corresponding OpenSearch plugins [installed on that cluster]({{site.url}}{{site.baseurl}}/opensearch/install/plugins/)
-- The corresponding version of [OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/) (e.g. OpenSearch Dashboards 1.0.0 works with OpenSearch 1.0.0)
+- The corresponding version of [OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/) (for example, OpenSearch Dashboards 1.0.0 works with OpenSearch 1.0.0)
```
Comment (Naarcha-AWS) [inline _dashboards/install/plugins.md:47]: I would change this to a more recent example, that includes at least a minor version:

```suggestion
- The corresponding version of [OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/) (for example, OpenSearch Dashboards 2.1.0 works with OpenSearch 2.1.0)
```

File: _dashboards/install/plugins.md [hunk 2]
```diff
@@ -20,25 +20,31 @@ If you don't want to use the all-in-one installation options, you can install th
...
+| Index Management | [index-management-dashboards-plugin](https://github.com/opensearch-project/index-management-dashboards-plugin) | 1.0.0 |
+| Notebooks | [dashboards-notebooks](https://github.com/opensearch-project/dashboards-notebooks) | 1.0.0 |
+| Notifications | [notifications](https://github.com/opensearch-project/notifications) | 2.0.0 |
+| Observability | [dashboards-observability](https://github.com/opensearch-project/observability/tree/main/dashboards-observability) | 2.0.0 |
+| Query Workbench | [query-workbench](https://github.com/opensearch-project/sql/tree/main/workbench) | 1.0.0 |
+| Reports | [dashboards-reports](https://github.com/opensearch-project/dashboards-reports) | 1.0.0 |
+| Security Analytics | [security-analytics-dashboards-plugin](https://github.com/opensearch-project/security-analytics-dashboards-plugin)| 2.4.0 |
+| Security | [security-dashboards-plugin](https://github.com/opensearch-project/security-analytics-dashboards-plugin) | 1.0.0 |
 
 ## Prerequisites
```
Comment (Naarcha-AWS) [inline _dashboards/install/plugins.md:43]: Should we move prereqs above plugin compatibility?

File: _dashboards/install/plugins.md [hunk 3]
```diff
@@ -20,25 +20,31 @@ If you don't want to use the all-in-one installation options, you can install th
...
-| queryWorkbenchDashboards | 1.0.0 or newer |
-| reportsDashboards | 1.0.0 or newer |
-| securityDashboards | 1.0.0 or newer |
-| traceAnalyticsDashboards | 1.0.0 or newer |
+Major, minor, and patch plugin versions must match OpenSearch major, minor, and patch versions in order to be compatible. For example, plugins versions 2.3.0.x work only with OpenSearch 2.3.0.
+{: .note}
+
+The following table lists available OpenSearch Dashboards plugins.
+
+| Plugin Name | Repository |Earliest Available Version |
```
Comment (Naarcha-AWS) [inline _dashboards/install/plugins.md:28]:
```suggestion
| Plugin Name | Repository | Earliest Available Version |
```

File: _opensearch/install/plugins.md [hunk 4]
```diff
@@ -205,56 +205,61 @@ bin/opensearch-plugin install --batch <plugin-name>
 
 ## Available plugins
 
+Major, minor, and patch plugin versions must match OpenSearch major, minor, and patch versions in order to be compatible. For example, plugins versions 2.3.0.x work only with OpenSearch 2.3.0.
```
Comment (Naarcha-AWS) [inline _opensearch/install/plugins.md:208]: I think we should either change this to `{: .important }` or add it as a warning? I'm afraid someone might miss it as a just a note.

File: _opensearch/install/plugins.md [hunk 5]
```diff
@@ -205,56 +205,61 @@ bin/opensearch-plugin install --batch <plugin-name>
...
-| Geospatial | [opensearch-geospatial](https://github.com/opensearch-project/geospatial) | 2.2.0 or newer |
-| Index Management | [opensearch-index-management](https://github.com/opensearch-project/index-management) | 1.0.0 or newer |
-| Job Scheduler | [opensearch-job-scheduler](https://github.com/opensearch-project/job-scheduler) | 1.0.0 or newer |
-| k-NN | [opensearch-knn](https://github.com/opensearch-project/k-NN) | 1.0.0 or newer |
-| ML Commons | [opensearch-ml](https://github.com/opensearch-project/ml-commons) | 1.3.0 or newer |
-| Observability | [opensearch-observability](https://github.com/opensearch-project/observability) | 1.2.0 or newer |
-| Performance Analyzer | [opensearch-performance-analyzer](https://github.com/opensearch-project/performance-analyzer) | 1.0.0 or newer |
-| Security | [opensearch-security](https://github.com/opensearch-project/security) | 1.0.0 or newer |
-| SQL | [opensearch-sql](https://github.com/opensearch-project/sql) | 1.0.0 or newer |
+| Dashboard Reports | [opensearch-reports-scheduler](https://github.com/opensearch-project/dashboards-reports) | 1.0.0 |
```
Comment (peterzhuamazon) [inline _opensearch/install/plugins.md:222]: The OS side of reports is called `Reports Scheduler`.

File: _opensearch/install/plugins.md (hunk 4 above)
Comment (kolchfa-aws) [reply _opensearch/install/plugins.md:208]: Important did not work, so changed to warning.

File: _dashboards/install/plugins.md [hunk 6]
```diff
@@ -21,31 +21,32 @@ If you don't want to use the all-in-one installation options, you can install th
...
+| Index Management Dashboards | [index-management-dashboards-plugin](https://github.com/opensearch-project/index-management-dashboards-plugin) | 1.0.0 |
+| Notebooks Dashboards | [dashboards-notebooks](https://github.com/opensearch-project/dashboards-notebooks) | 1.0.0 |
+| Notifications Dashboards | [notifications](https://github.com/opensearch-project/notifications) | 2.0.0 |
+| Observability Dashboards | [dashboards-observability](https://github.com/opensearch-project/observability/tree/main/dashboards-observability) | 2.0.0 |
+| Query Workbench Dashboards | [query-workbench](https://github.com/opensearch-project/sql/tree/main/workbench) | 1.0.0 |
+| Reports Dashboards | [dashboards-reports](https://github.com/opensearch-project/dashboards-reports) | 1.0.0 |
+| Security Analytics Dashboards | [security-analytics-dashboards-plugin](https://github.com/opensearch-project/security-analytics-dashboards-plugin)| 2.4.0 |
+| Security Dashboards | [security-dashboards-plugin](https://github.com/opensearch-project/security-analytics-dashboards-plugin) | 1.0.0 |
```
Comment (peterzhuamazon) [inline _dashboards/install/plugins.md:49]: Missing:
`  - name: customImportMapDashboards
`
Same time as geospacial os plugin

File: _opensearch/install/plugins.md [hunk 7]
```diff
@@ -219,7 +219,7 @@ The following plugins are bundled with all OpenSearch distributions except for m
 | Asynchronous Search | [opensearch-asynchronous-search](https://github.com/opensearch-project/asynchronous-search) | 1.0.0 |
 | Cross Cluster Replication | [opensearch-cross-cluster-replication](https://github.com/opensearch-project/cross-cluster-replication) | 1.1.0 |
 | Dashboard Notebooks<sup>1</sup> | [opensearch-notebooks](https://github.com/opensearch-project/dashboards-notebooks) | 1.0.0 to 1.1.0 |
-| Dashboard Reports | [opensearch-reports-scheduler](https://github.com/opensearch-project/dashboards-reports) | 1.0.0 |
+| Reports Scheduler | [opensearch-reports-scheduler](https://github.com/opensearch-project/dashboards-reports) | 1.0.0 |
```
Comment (peterzhuamazon) [inline _opensearch/install/plugins.md:222]: Also missing `notifications` as it has a OS part as well.

**Review by ariamarble — APPROVED** [review]
LGTM other than comments already left

**Review by peterzhuamazon — APPROVED** [review]

**Review by alicejw-aws — APPROVED** [review]
LGTM

**Review by Naarcha-AWS — APPROVED** [review]

**Pull request merged**